weather/rest.go

```diff
@@ -83,6 +83,9 @@ func RestGetHandler(res http.ResponseWriter, req *http.Request) {
 		}
 		res.Header().Set("Content-Type", "application/json")
 		res.WriteHeader(http.StatusOK)
-		json.NewEncoder(res).Encode(output)
+		err := json.NewEncoder(res).Encode(output)
```

Comment:
If we are going to handle this error,  then we should find a way to handle it such that failures don't occur after we have already written the response code to the http response

Reply:
Touche, salesman!